test/end-to-end/core.sh

```diff
@@ -48,6 +48,7 @@ function wait_for_app() {
 
 # service dns entry is visible via master service
 # find the IP of the master service by asking the API_HOST to verify DNS is running there
+// TODO: use clusterDomain instead of cluster.local
```

Comment:
I think we would leave end-to-end with the default. might be worth an extended test for the custom domain setup